### DISCUSSION TITLE
Handling multiple years' forms

### DISCUSSION THREAD

**Discussion by zakpatterson** [ustaxes/UsTaxes#703] [Ideas]
A discussion started on discord that might lead to significant changes for the project.

We have the problem of how to represent multiple years of tax forms. One proposal is to separate the tax form models into a separate project, with each year represented by a parallel master branch. Each year could have its own release flow. 

Migration steps: 

* Split form models into separate project, as branch `main-2020`. Start branches `main-2019`, `main-2021` including revisions to all forms pdfs via git patches.
* Each year will have its own release line.
* The released packages will all be included in the root USTaxes project as dependencies
* Root project should be able to access each library of form by year and generate the appropriate year based on what the user is working on.

2021 tax forms
---
Draft tax forms are currently available.

https://www.irs.gov/pub/irs-dft/f1040s3--dft.pdf

**Comment by thegrims** [comment]
I think this makes a lot of sense for separation of concerns. If the tax year packages are published on NPM as well, others might be able to use them and contribute.

**Comment by thegrims** [comment]
Does this have any implications for #662? @zakpatterson

**Reply by zakpatterson** [reply]
I've started https://github.com/ustaxes/ustaxes-forms and I'm looking at using it in that PR.
